src/rules/checkTagNames.js

```diff
@@ -54,12 +110,103 @@ export default iterateJsdoc(({
       });
   }
 
-  for (const jsdocTag of jsdoc.tags) {
+  const isInAmbientContext = (subNode) => {
+    return subNode.type === 'Program' ?
+      context.getFilename().endsWith('.d.ts') :
+      Boolean(subNode.declare) || isInAmbientContext(subNode.parent);
+  };
+
+  const tagIsRedundantWhenTyped = (jsdocTag) => {
+    if (!typedTagsUnnecessaryOutsideDeclare.has(jsdocTag.tag)) {
+      return false;
+    }
+
+    if (jsdocTag.description.trim() === 'default' || jsdocTag.name === 'default') {
```

Comment:
With `default` as a tag description, is this some recognized TS behavior? I'd rather we not add features for idiosyncratic conventions.

Reply:
I'll defer to @sandersn as dtslint person - but I believe this is because people tend to do things like:

```ts
/** @default 0 */
let value = 0;
```

Reply:
I'm specifically asking about `jsdocTag.description.trim() === 'default'`, i.e.,. where `default` is in the description area like this example:

```js
/** @abstract - default */
```

Reply:
Yeah good point, I don't know why this would be the case. Pending information from @sandersn / anybody with more context, I'll remove that point. Thanks!